inc/Engine/Optimization/RUCSS/Warmup/Status/Checker.php

```diff
@@ -45,31 +45,108 @@ public function __construct( Options $options_api, Options_Data $options, Resour
 		$this->resources_query = $resources_query;
 	}
 
+
 	/**
-	 * Checks warmup status for resources in the prewarmup process
+	 * Update Warmup process on completion.
 	 *
 	 * @return void
 	 */
-	public function check_warmup_status() {
+	private function set_warmup_status_completed() {
+		$this->set_warmup_status_finish_time();
+		$this->set_warmup_force_optimization();
+	}
+
+	/**
+	 * Automatically stop warmup process after 1 hour in case the process did not finished.
+	 *
+	 * @return void
+	 */
+	public function auto_stop_warmup_after_1hour() {
+		if ( $this->is_warmup_finished() ) {
+			return;
+		}
+
 		$prewarmup_stats = $this->options_api->get( 'prewarmup_stats', [] );
 
-		if ( empty( $prewarmup_stats['fetch_finish_time'] ) ) {
+		if ( empty( $prewarmup_stats['scan_start_time'] ) ) {
 			return;
 		}
 
 		if ( time() > strtotime( '+1 hour', (int) $prewarmup_stats['scan_start_time'] ) ) {
-			$this->set_warmup_status_finish_time();
+			$this->set_warmup_status_completed();
+		}
+	}
+
+	/**
+	 * Checks if warmup is completed.
```

Comment:
This description is misleading. It is activating the optimization and setting the finish time on completion.